src/coredata.cs

```diff
@@ -2677,6 +2725,43 @@ interface NSPersistentCloudKitContainer {
 		[Watch (7,0), TV (14,0), Mac (11,0), iOS (14,0)]
 		[Export ("canModifyManagedObjectsInStore:")]
 		bool CanModifyManagedObjects (NSPersistentStore store);
+
+		// NSPersistentCloudKitContainer_Sharing
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerAcceptShareInvitationsResult")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("acceptShareInvitationsFromMetadata:intoPersistentStore:completion:")]
+		void AcceptShareInvitations (CKShareMetadata[] metadata, NSPersistentStore persistentStore, [NullAllowed] NSPersistentCloudKitContainerAcceptShareInvitationsHandler handler);
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerPurgeObjectsAndRecordsInZone")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("purgeObjectsAndRecordsInZoneWithID:inPersistentStore:completion:")]
+		void PurgeObjectsAndRecordsInZone (CKRecordZoneID zoneId, [NullAllowed] NSPersistentStore persistentStore, [NullAllowed] NSPersistentCloudKitContainerPurgeObjectsAndRecordsInZoneHandler handler); 
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerPersistUpdatedShareResult")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("persistUpdatedShare:inPersistentStore:completion:")]
+		void PersistUpdatedShare (CKShare share, NSPersistentStore persistentStore, [NullAllowed] NSPersistentCloudKitContainerPersistUpdatedShareHandler handler);
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerFetchParticipantsMatchingLookupInfosResult")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("fetchParticipantsMatchingLookupInfos:intoPersistentStore:completion:")]
+		void FetchParticipantsMatchingLookupInfos (CKUserIdentityLookupInfo[] lookupInfos, NSPersistentStore persistentStore, NSPersistentCloudKitContainerFetchParticipantsMatchingLookupInfosHandler handler); 
+
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("fetchSharesMatchingObjectIDs:error:")]
+		[return: NullAllowed]
+		NSDictionary<NSManagedObjectID, CKShare> FetchSharesMatchingObjectIds (NSManagedObjectID[] objectIDs, [NullAllowed] out NSError error);
+
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("fetchSharesInPersistentStore:error:")]
+		[return: NullAllowed]
+		CKShare[] FetchSharesInPersistentStore ([NullAllowed] NSPersistentStore persistentStore, [NullAllowed] out NSError error);
+
+		[Async (ResultTypeName = "NSPersistentCloudKitContainerShareManagedObjectsResult")]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("shareManagedObjects:toShare:completion:")]
+		void ShareManagedObjects (NSManagedObject[] managedObjects, [NullAllowed] CKShare share, NSPersistentCloudKitContainerShareManagedObjectsHandler handler);
```

Comment:
same

Reply:
All this, I got th first interface wrong and forgot to update the delegates, updated in https://github.com/xamarin/xamarin-macios/pull/12183/commits/a9c90671922a8fac77fa8580605bae1715fdf4ed